core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/tenant/JDBCTenantManager.java

```diff
@@ -1039,6 +1038,30 @@ private void invalidateCacheManager(String domain) {
         cacheManagerFactory.removeCacheManagerMap(domain);
     }
 
+    private void deleteUMTenantByTenantUUID(String tenantUUID, Connection connection) throws SQLException {
+        String sqlStmt = TenantConstants.DELETE_TENANT_BY_UUID_SQL;
+        try (PreparedStatement prepStmt = connection.prepareStatement(sqlStmt)) {
```

Comment:
Can't we create a DB connection inside the method?

Reply:
if we do so, we cannot delete UM_TENANT and UM_DOMAIN tables at once. Rollbacks happen separately for each table if an error occurs.